server/src/main/java/org/elasticsearch/index/SearchSlowLog.java

```diff
@@ -174,6 +174,11 @@ public String toString() {
             } else {
                 sb.append("source[], ");
             }
+            if (context.getTask().getHeader("X-Opaque-Id") != null) {
+                sb.append("id[").append(context.getTask().getHeader("X-Opaque-Id")).append("], ");
```

Comment:
Since the header name is used in several places throughout the code base, maybe it makes sense to declare it as a public string constant somewhere? I see its already a package private constant in DefaultRestChannel, maybe it should be public, get some proper comment etc... so we can use it here (and in other locations).